### DISCUSSION TITLE
wsl.exe <cmd> doesn't run shell or profile scripts

### DISCUSSION THREAD

**Discussion by Spongman** [microsoft/WSL#6250] [General] [answered]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19041.610]
Your Distribution version: Centos7
Whether the issue is on WSL 2 and/or WSL 1: Linux version 4.19.128-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Tue Jun 23 12:58:10 UTC 2020
```

# Steps to reproduce
- `wsl.exe htop`
![image](https://user-images.githubusercontent.com/1088194/99438243-62bc2600-28c8-11eb-8755-5293bbbfbb0a.png)

wsl is running the bare command without running the user's login shell. this means that the user's profile / rc scripts are not run and the environment is not correctly configured.

i have scripts in my `.bashrc` for enabling packages installed in `/opt/rh`. not running these causes everything to break. for example, vscode remote can't run properly because it can't find `git`, which is enabled by these scripts.

`sshd`, for example uses the user's login shell to run commands causing the environment to be correctly configured. wsl should do the same.

here's `wsl.exe env` :
```
HELL=/bin/bash
TERM=xterm-256color
WSLENV=
USER=user
NAME=machine
WSL_DISTRO_NAME=CentOS7
HOSTTYPE=x86_64
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/WINDOWS/system32:/mnt/c/WINDOWS:<rest of windows path>
PWD=/mnt/c/Users/User
LANG=en_US.UTF-8
WSL_INTEROP=/run/WSL/20099_interop
SHLVL=1
HOME=/home/user
LOGNAME=user
_=/usr/bin/env
```

where is it even getting `/usr/local/games` from? that's not in my PATH anywhere...

**Comment by 0xbadfca11** [comment, marked as answer]
It is the same behavior as Linux. It's not an interactive shell.
https://www.gnu.org/software/bash/manual/html_node/Bash-Startup-Files.html
#4955 #4740 #2816 #2075
